### ISSUE TITLE
Exception thrown while executing UI Block: Collection was being mutated while being enumerated

### ISSUE THREAD

**Issue by baltagih2** [TheWidlarzGroup/rn-emoji-keyboard#107]
**Describe the bug**
After following the docs, installing rn-emoji-keyboard and adding it to my project, when selecting an emoji from the keyboard I get this error (screenshot below). This sometimes does not happen the first time I select an emoji but it always happens in the first 3 selections.

**To Reproduce**
`
import EmojiPicker from "rn-emoji-keyboard";

export default function MyEmojiPicker({isOpen, onClose, onSelect}) {
<EmojiPicker
        onEmojiSelected={(e) => onSelect(e.emoji)}
        open={isOpen}
        onClose={onClose}
 />
}

`

**Expected behavior**
The keyboard is supposed to not crash the entire app when an emoji is selected

**Screenshots**
![Simulator Screen Shot - iPhone 14 Pro - 2022-12-22 at 19 05 39](https://user-images.githubusercontent.com/62379853/209189153-6bf52b59-c892-48ff-8c0e-8b6cf891d4b9.png)

**Devices this happens on (that I personally tested)**
- Simulator iPhone 14 pro - iOS 16.2
- Simulator Resizable API 32 - Android 12L
- Simulator Pixel 6 Pro API Tiramisu - Android 13
- Real device iPhone14 pro - iOS 16.2
- Real device iPhone 11 pro - iOS 16.1
- Real device Huawei Y9 - Android 12

**Stack & Versions**
"expo": "^47.0.0",
"react": "18.1.0",
"react-native": "0.70.5",
"rn-emoji-keyboard": "^1.1.0"

**Comment by mateki0** [comment]
Hey @baltagih2, I've tried to reproduce it on iPhone 11 and 14 pro simulator (iOS 16.2) and I didn't get any crash. Do you have react-native-reanimated in your project and if so in which version? Do you have any other packages installed that may cause this bug or these are all?

**Comment by baltagih2** [comment]
We are using `"react-native-reanimated": "~2.12.0"` on expo SDK 47 with React native gifted chat `"react-native-gifted-chat": "^1.1.0"`.

The error sometimes only happens after the second or third selection, I'm not sure if it's specific emojis or if the problem is somewhere else.

Any thing I can do to help you debug?

**Comment by mateki0** [comment]
Can you paste your whole component here? It may help, I saw this problem but it was connected with modal animation and entering/exiting animations from reanimated. I have the same versions of the packages and everything works fine so I guess it must be something with the component code.

**Comment by baltagih2** [comment]
Here's a minimal version of my code in which the problem still occurs. The "EmojiKeyboard" component is just the "EmojiPicker" wrapped in an ErrorBoundary. When "handleLongPress" is called, an emoji drawer appears with a "plus" sign (similar to instagram) and when the plus sign is pressed, it sets emojiKeyOpen to true to open the emoji keyboard.

`export default function MainChatBubble({
  props,
}: {
  props: BubbleProps<ChatMessage>;
}) {
  return (
    <View>
      <GestureHandlerRootView>
        <Swipeable
          ref={swipeableRef}
          friction={2}
          renderRightActions={() =>
            isRight ? <View style={{ width: 16 }} /> : null
          }
          renderLeftActions={() =>
            !isRight ? <View style={{ width: 16 }} /> : null
          }
          onSwipeableOpen={handleSwipe}
        >
          <Bubble {...props} onLongPress={handleLongPress} />
        </Swipeable>
      </GestureHandlerRootView>
      <EmojiKeyboard
        isOpen={emojiKeyOpen}
        onClose={() => {
          setEmojiKeyOpen(false);
          setShowEmojiesSelector((prev) => false);
        }}
        onSelect={(e) => handleSelectEmoji(e)}
      />
    </View>
  );
}`

**Comment by mateki0** [comment]
Thanks for the code. Can I also see the "emoji drawer"? And when do you close this drawer?

**Comment by baltagih2** [comment]
Here's the EmojiKeyboard code:

```
export default function EmojiKeyboard({ isOpen, onClose, onSelect }: Props) {
  return (
    <ErrorBoundary onClose={onClose}>
      <EmojiPicker
        onEmojiSelected={(e: { emoji: string }) => onSelect(e.emoji)}
        open={isOpen}
        onClose={onClose}
      />
    </ErrorBoundary>
  );
}
```

This is a custom Error Boundary to handle the error thrown by the EmojiPicker so the app does not crash and it's passed the onClose function to close the drawer if an error occurs so the user can continue chatting as usual.

setShowEmojiesSelector((prev) => false) is responsible for closing the small emoji popup that allows the user to select an emoji quickly (a couple of most used emojis displayed). This popup contains a "plus" sign which handles opening the emojiPicker

**Comment by mateki0** [comment]
Can you check if it works well without this small popup? Sorry for so many questions, but I still can't reproduce it and I'm trying to get as small piece of code as it's possible :)

**Comment by mateki0** [comment]
It may be the case, I think that the safiest option is to use the same version as we have in the library if it's possible. But anyway emojis are not svg, only the categories icons are, so it's weird. Without ErrorBoundary it is the same?

**Comment by mateki0** [comment]
Okay, unfortunately I still can't reproduce it on my computer, but I'll talk about it with @jakex7 in the next week and maybe he  will be able to see the error on his computer.

**Comment by baltagih2** [comment]
Thank you so much for all the effort, I'll keep trying different things on my end to see if I can find the root cause

**Comment by mateki0** [comment]
Let me know if you come up with anything new and I'll take a look again :)

**Comment by baltagih2** [comment]
Hey! Thank you for your help. I'm still running into the same problem and can't seem to get it to work. I also took the keyboard outside of the bubble component into the main one and the problem still occurs. Anything I can try to give you more visibility into the error or more details by any chance?

**Comment by mateki0** [comment]
Maybe you can create a snack with minimal code needed to reproduce it? For example using this tool - https://snack.expo.dev/
Then we will know if it's something with your project only, or no :)

**Comment by baltagih2** [comment]
The error happens in this snack that I created: https://snack.expo.dev/@baltagih/intrigued-candies
I tried to remove as much of the clutter as possible. Let me know if you need me to reduce it even more.

Instructions to reproduce error:
- Long press on any of the messages
- An emoji selector will appear, click on the "plus" sign on the right
- The emoji keyboard will appear
- Select an emoji (sometimes the error happens when you select a second one or when you choose and emoji from one of the less-used categories like the airplane)
- if no errors are displayed, for you to know that the error happened, the small emoji selector will appear empty (Looks like it crashes all svgs or something) and you won't be able to select a second emoji anymore


Tips:
- The rn-emoji-keyboard is available in the MainChatBubble.js component at the bottom
- The App.js component is just the GiftedChat wrapped in a view and default constant messages are passed to it

Let me know if I can help with anything else in debugging this and thank you again for all your help!

**Comment by mateki0** [comment]
Hi @baltagih2, can you try to move `setShowEmojiesSelector((prev) => false);` from the handleSelectEmoji fn and use it in '+' onPress? Also you can remove 'setShowEmojiesSelector((prev) => false)' from the emoji onClose function. I think it works fine then, but still it's only snack, and it would be great to test it in your app.

**Comment by baltagih2** [comment]
Looks like this fixed it. The small emoji selector with the + sign doesn't close anymore on selection, but at least I can't replicate the crash anymore. Any idea why this could be happening? I'll find a way to keep the same functionality with this setup but it's weird that closing the small selector makes the app crash

**Comment by mateki0** [comment]
I'm not sure why this happens. All I found is [this thread](https://github.com/software-mansion/react-native-reanimated/issues/3286) and it says that there are some bugs with layout transitions. However after my changes the small emoji selector should close right after clicking +, so there's no reason to close it after selection. I've added ten emojis to one message and it was fine. 

Great, let me know if it will crash again in your app :)
